pkg/codegen/internal/test/testdata/simple-plain-schema/dotnet/Component.cs

```diff
@@ -67,6 +73,17 @@ public sealed class ComponentArgs : Pulumi.ResourceArgs
         [Input("b")]
         public bool? B { get; set; }
 
+        [Input("bar")]
+        public Inputs.Foo? Bar { get; set; }
+
+        [Input("baz")]
+        private ImmutableArray<Inputs.Foo>? _baz;
+        public ImmutableArray<Inputs.Foo> Baz
+        {
+            get => _baz ?? (_baz = new ImmutableArray<Inputs.Foo>());
```

Comment:
This has uncovered an orthogonal bug. (I'll open a separate issue and work on a fix). Compiler error:

```
Cannot implicitly convert type 'System.Collections.Immutable.ImmutableArray<Pulumi.Example.Inputs.Foo>?' to 'System.Collections.Immutable.ImmutableArray<Pulumi.Example.Inputs.Foo>'. An explicit conversion exists (are you missing a cast?)
```

I think because `System.Collections.Immutable.ImmutableArray<T>` is a struct.

Reply:
https://github.com/pulumi/pulumi/issues/6813